core/types/types.cc

```diff
@@ -843,17 +848,12 @@ core::SymbolRef Types::getRepresentedClass(core::Context ctx, const core::Type *
         singleton = s->symbol;
     } else {
         auto *at = core::cast_type<core::AppliedType>(ty);
-        if (at == nullptr) {
-            return core::Symbols::noSymbol();
-        }
-        // If this class is a "real" generic, leave it alone. We're not quite
-        // sure yet what it means to move between the "attached" and "singleton"
-        // levels for generic singletons. However, if the singleton has kind `*`
-        // due to all type members being fixed:, we're OK to treat the
-        // AppliedType and the Symbol as interchangeable here.
-        if (at->klass.data(ctx)->typeArity(ctx) != 0) {
+
+        // TODO: should the singletonClass check be an ENFORCE?
```

Comment:
I didn't understand this `TODO`

Reply:
It wasn't clear to me if we would ever see a non-singleton class at this point. I was wondering if the check for the `klass` being a singleton would be something that we could just `ENFORCE` rather than check.

Reply:
+1 for enforcing it :-)

Reply:
Maybe the right place to move the ENFORCE to is right before the call to `attachedClass`. That way we'd be enforcing for `ClassType` as well.